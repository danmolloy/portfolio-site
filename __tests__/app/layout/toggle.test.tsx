import ToggleSwitch from "@/app/layout/toggle";
import { render, act, fireEvent, screen } from "@testing-library/react";





describe("<ToggleSwitch />", () => {
  const mockLocalStorage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    length: 0, 
    clear: jest.fn(), 
    key: jest.fn(), 
  };
  
  window.localStorage = mockLocalStorage; // Mock localStorage behavior
  
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
  beforeEach(() => {
    render(<ToggleSwitch />)
  })
  it("toggle-switch is in the document", () => {
    const toggle = screen.getByTestId("toggle-switch")
    expect(toggle).toBeInTheDocument()
  })
  it("updates to dark mode on click", () => {
    const toggle = screen.getByTestId("toggle-switch")
    act(() => {
      fireEvent.click(toggle)
    })
    expect(window.matchMedia).toHaveBeenCalledWith("(prefers-color-scheme: dark)")
    act(() => {
      fireEvent.click(toggle)
    })
    expect(window.matchMedia).toHaveBeenCalledWith("(prefers-color-scheme: dark)")

  })
  it("matches snapshot", () => {
    const toggle = screen.getByTestId("toggle-switch")
    expect(toggle).toMatchSnapshot()
  })
})

