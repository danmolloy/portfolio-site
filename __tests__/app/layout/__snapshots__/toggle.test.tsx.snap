// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> matches snapshot 1`] = `
<div
  class="hover:cursor-pointer flex flex-row flex-wrap items-center  my-2 mx-4"
  data-testid="toggle-switch"
>
  <div
    class=" overflow-y-visible w-8 h-4 flex flex-row items-center rounded-full bg-zinc-50 border"
  >
    <div
      class=" -ml-2 bg-zinc-100 border shadow w-5 h-5 rounded-full transition-all duration-500"
    />
  </div>
      
</div>
`;
